scalyr_agent/agent_main.py

```diff
@@ -1925,7 +1944,19 @@ def __report_status_to_file(self):
             tmp_file = None
 
             os.rename(tmp_file_path, final_file_path)
-        except (OSError, IOError):
+        except (OSError, IOError) as e:
+            # Temporary workaround to make race conditions less likely.
+            # If agent status or health check is requested multiple times or concurrently around the
+            # same time, it's likely the race will occur and rename will fail because the file  was
+            # already renamed by the other process.
+            # This workaround is not 100%, only 100% solution is to use a global read write lock
+            # and only allow single invocation of status command at the same time.
+            if tmp_file is not None:
+                tmp_file.close()
+
+            if os.path.isfile(final_file_path):
+                return final_file_path
```

Comment:
Is returning something here right?  I don't think this method has a return value.

Reply:
Yep, that method does have a return value - https://github.com/scalyr/scalyr-agent-2/blob/f952e8fbeae7d6937245feb210ca207f8df91233/scalyr_agent/agent_main.py#L1971.

On that note, that's where type annotations and MyPy would also come handy - if method didn't declare a return, but we returned something, mypy would catch that.